src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -30,178 +31,125 @@ import { WithDetailsReceipt } from '@/components/WithDetailsReceipt';
+  } = orEmptyObj(validationResponse?.summaryData);
+
+  // save initial type to state if it is provided
+  useEffect(() => {
+    if (initialType == null) {
+      return;
+    }
+    dispatch(setVaultFormOperation(initialType));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   const setAmount = (change: NumberFormatValues) => {
-    setAmountState(change.value);
+    dispatch(setVaultFormAmount(change.value));
+  };
+  const setAmountState = (newVal: string) => {
+    dispatch(setVaultFormAmount(newVal));
+  };
+  const setOperation = (op: 'DEPOSIT' | 'WITHDRAW') => {
+    dispatch(setVaultFormOperation(op));
   };
 
-  const errors = useMemo((): VaultFormError[] => {
-    if (!canViewAccount) {
-      return [
-        {
-          type: AlertType.Error,
-          key: 'disconnected',
-          short: stringGetter({ key: STRING_KEYS.CONNECT_WALLET }),
-        },
-      ];
-    }
-    if (isAccountViewOnly) {
-      return [
-        {
-          type: AlertType.Error,
-          key: 'view-only',
-          short: stringGetter({ key: STRING_KEYS.NOT_ALLOWED }),
-        },
-      ];
-    }
-    if (MustBigNumber(amount).eq(0)) {
-      return [
-        {
-          type: AlertType.Error,
-          key: 'amount-empty',
-          short:
-            selectedType === 'deposit'
-              ? stringGetter({ key: STRING_KEYS.ENTER_AMOUNT_TO_DEPOSIT })
-              : stringGetter({ key: STRING_KEYS.ENTER_AMOUNT_TO_WITHDRAW }),
-        },
-      ];
-    }
-    const allErrors: VaultFormError[] = [];
-    if (selectedType === 'deposit') {
-      if (freeCollateralUpdated == null || freeCollateralUpdated < 0) {
-        allErrors.push({
-          type: AlertType.Error,
-          key: 'deposit-high',
-          long: stringGetter({ key: STRING_KEYS.DEPOSIT_TOO_HIGH }),
-          short: 'Modify amount',
-        });
-      }
-    } else {
-      if (userBalanceUpdated == null || userBalanceUpdated < 0) {
-        allErrors.push({
-          type: AlertType.Error,
-          key: 'withdraw-high',
-          long: stringGetter({ key: STRING_KEYS.WITHDRAW_TOO_HIGH }),
-          short: 'Modify amount',
-        });
-      }
-      if (slippagePercent >= SLIPPAGE_PERCENT_WARN) {
-        allErrors.push({
-          type: AlertType.Warning,
-          key: 'slippage-high',
-          long: (
-            <span>
-              {stringGetter({
-                key: STRING_KEYS.SLIPPAGE_WARNING,
-                params: {
-                  AMOUNT: <$InlineOutput value={slippagePercent} type={OutputType.Percent} />,
-                  LINK: (
-                    <Link href={vaultsLearnMore} withIcon isInline>
-                      {stringGetter({ key: STRING_KEYS.VAULT_FAQS })}
-                    </Link>
-                  ),
-                },
-              })}
-            </span>
-          ),
+  const errors = useMemo(
+    () =>
+      validationResponse?.errors.toArray().map((error) => {
```

Comment:
can `toArray()` be called within the validationResponse hook? It would be nice to convert Abacus' `iList` before usage, but I know this is nested within `errors`.

Reply:
oof. Okay makes sense